test/unittests/validation_test.cpp

```diff
@@ -129,6 +129,149 @@ TEST(validation, memory_size_no_memory)
         parse(wasm), validation_error, "memory instructions require imported or defined memory");
 }
 
+TEST(validation, store8_invalid_alignment)
```

Comment:
I see some issues with tests:
- we don't have error location information so it is unclear which `align=` is causing the exception,
- positive tests are missing, we should add ones for max alignment, e.g `i32.store8 align=1`, `i32.store16 align=2`, etc.
- Is `align=0` ok?

Reply:
Yes, test should better be split to valid cases / one invalid instruction per case.

Reply:
I tried to avoid having >100 test cases. Any suggestions?

Reply:
In section 4.4.4:
> Note: The alignment memarg.align in load and store instructions does not affect the semantics. It is an indication that the offset ea at which the memory is accessed is intended to satisfy the property ea mod 2**memarg.align = 0. A WebAssembly implementation can use this hint to optimize for the intended use. Unaligned access violating that property is still allowed and must succeed regardless of the annotation. However, it may be substantially slower on some hardware.

So `align=0` would mean `address % 1 == 0` must be satisfied for aligned access. To me it suggests align must be in the range of [0, 3] (as seen in the comment in the parser).

Reply:
@gumb0 @chfast happy with these kind of tests (check `store8_alignment`)? I can go through the pain of doing it for every option... will be probably a few hundred lines.